stan/math/prim/prob/multi_normal_lpdf.hpp

```diff
@@ -71,35 +90,173 @@ return_type_t<T_y, T_loc, T_covar> multi_normal_lpdf(const T_y& y,
+      }
+    } else {
+      matrix_partials_t inv_Sigma
+          = mdivide_left_ldlt(ldlt_Sigma, Eigen::MatrixXd::Identity(K, K));
+
+      half = (inv_Sigma * y_val_minus_mu_val);
+
+      logp += 0.5 * sum(log(D_inv)) * size_vec;
+
+      partials<2>(ops_partials) += -0.5 * size_vec * inv_Sigma;
+
+      for (size_t i = 0; i < size_vec; i++) {
+        partials_vec<2>(ops_partials)[i] += 0.5 * half.col(i) * half.row(i);
+      }
+    }
+
+    logp += -0.5 * y_val_minus_mu_val.cwiseProduct(half).sum();
+
+    for (size_t i = 0; i < size_vec; i++) {
+      if (!is_constant_all<T_y>::value) {
+        partials_vec<0>(ops_partials)[i] -= half.col(i);
+      }
+      if (!is_constant_all<T_loc>::value) {
+        partials_vec<1>(ops_partials)[i] += half.col(i);
+      }
+    }
+  }
+
+  return ops_partials.build(logp);
+}
+template <bool propto, typename T_y, typename T_loc, typename T_covar,
+          require_all_vector_vt<is_stan_scalar, T_y, T_loc>* = nullptr,
+          require_all_not_nonscalar_prim_or_rev_kernel_expression_t<
+              T_y, T_loc, T_covar>* = nullptr>
+return_type_t<T_y, T_loc, T_covar> multi_normal_lpdf(const T_y& y,
+                                                     const T_loc& mu,
+                                                     const T_covar& Sigma) {
+  using T_covar_elem = typename scalar_type<T_covar>::type;
+  using T_return = return_type_t<T_y, T_loc, T_covar>;
+  using T_partials_return = partials_return_t<T_y, T_loc, T_covar>;
+  using matrix_partials_t
+      = Eigen::Matrix<T_partials_return, Eigen::Dynamic, Eigen::Dynamic>;
+  using vector_partials_t = Eigen::Matrix<T_partials_return, Eigen::Dynamic, 1>;
+  using row_vector_partials_t
+      = Eigen::Matrix<T_partials_return, 1, Eigen::Dynamic>;
+  using T_y_ref = ref_type_t<T_y>;
+  using T_mu_ref = ref_type_t<T_loc>;
+  using T_Sigma_ref = ref_type_t<T_covar>;
+
+  static const char* function = "multi_normal_lpdf";
+  check_positive(function, "Covariance matrix rows", Sigma.rows());
+
+  T_y_ref y_ref = y;
+  T_mu_ref mu_ref = mu;
+  T_Sigma_ref Sigma_ref = Sigma;
+
+  decltype(auto) y_val = as_value_column_vector_or_scalar(y_ref);
+  decltype(auto) mu_val = as_value_column_vector_or_scalar(mu_ref);
+
+  const int size_y = y_ref.size();
+  const int size_mu = mu_ref.size();
+  const unsigned int K = Sigma.rows();
+
+  check_finite(function, "Location parameter", mu_val);
+  check_not_nan(function, "Random variable", y_val);
+
+  check_size_match(function, "Size of random variable", size_y,
+                   "size of location parameter", size_mu);
+  check_size_match(function, "Size of random variable", size_y,
+                   "rows of covariance parameter", Sigma.rows());
+  check_size_match(function, "Size of random variable", size_y,
+                   "columns of covariance parameter", Sigma.cols());
+
+  check_symmetric(function, "Covariance matrix", Sigma_ref);
+
+  auto ldlt_Sigma = make_ldlt_factor(Sigma_ref);
```

Comment:
I think this line is where you're getting vars that lead to an error later. At least for the first derivative, the `partial_return_t` type is going to be `double`, but `Sigma_ref` might be a `var`. 

`auto ldlt_Sigma = make_ldlt_factor(value_of(Sigma_ref));` gets your test compiling (and may even be correct?).

Reply:
It looks like every single place we call `make_ldlt_factor` we use auto, so I doubt we're the only ones who don't know how to name it's type generically